apps/web/components/Breadcrumbs.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+import { usePathname } from "next/navigation";
+import Link from "next/link";
+import React, { useEffect, useMemo, useState } from "react";
+import { ChevronRightIcon } from "@heroicons/react/24/solid";
+import { getTitlesFromUrlSegments } from "@/services/getTitlesFromUrlSegments";
+
+interface Breadcrumb {
+  href: string;
+  label: string;
+}
+
+const truncateString = (str: string) => {
+  return `${str.slice(0, 6)}...${str.slice(-4)}`;
+};
+
+export function Breadcrumbs() {
+  const path = usePathname();
+  const [breadcrumbs, setBreadcrumbs] = useState<Breadcrumb[]>([]);
+
+  const breadcrumbsPromise = useMemo(async (): Promise<Breadcrumb[]> => {
+    const segments = path.split("/").filter((segment) => segment !== "");
+
+    const titles = await getTitlesFromUrlSegments(segments);
+
+    return segments
+      .map((segment, index) => {
+        if (index < 2) {
+          return undefined;
+        }
+
+        const href = `/${segments.slice(0, index + 1).join("/")}`;
+        let displayLabel = segment.startsWith("0x")
+          ? truncateString(segment)
+          : segment;
+
+        if (!!titles) {
+          // index correction as first 2 segments are /gardens/[chainId]
+          const title = titles[index - 2];
+          if (!!title) displayLabel = title;
+        }
+        return { href, label: displayLabel };
+      })
+      .filter((segment): segment is Breadcrumb => segment !== undefined);
+  }, [path]);
+
+  useEffect(() => {
+    (async () => {
+      const result = await breadcrumbsPromise;
+      setBreadcrumbs(result);
+    })();
+  }, [breadcrumbsPromise]);
+
+  if (!breadcrumbs.length) return null;
+
+  return (
+    <div aria-label="Breadcrumbs" className="flex items-center justify-center">
+      <ol className="flex items-center justify-center">
+        {breadcrumbs.map(({ href, label }, index) => (
+          <li
```

Comment:
Maybe add an ellipsis: 
![image](https://github.com/1Hive/gardens-v2/assets/13890618/70b7470c-56c3-477e-9ec4-462565f0f958)
https://gardens-v2-git-implement-breadcrumbs-1hive.vercel.app/gardens/421614/0xcc6c8b9f745db2277f7aac1bc026d5c2ea7bd88d/0x535f7dc4916ee558f8d6600adb067224d4a72254

Reply:
whats an ellipsis? should i just truncate the string? like max cap it to 30 chars or something and then add "..."?

Reply:
That is what ellipsis is. Here some example https://www.npmjs.com/package/react-lines-ellipsis but maybe implement manually for you guys make more sense

Reply:
or just use css

Reply:
text-ellipsis I think in tailwind

Reply:
![Screenshot from 2024-07-09 12-12-48](https://github.com/1Hive/gardens-v2/assets/54861577/b7eb0a51-d6d4-469e-8bf5-d099e0dc452f)
implemented